utils/supabase/queries.ts

```diff
@@ -37,3 +43,24 @@ export const getUserDetails = cache(async (supabase: SupabaseClient) => {
     .single();
   return userDetails;
 });
+
+export const getUserSubscriptions = cache(async (supabase: SupabaseClient, user: User | null) => {
```

Comment:
This function doesn't seem necessary. When there are multiple subscriptions, the UI should generate an error to let the user know, because multiple subscriptions don't make sense to users.